### ISSUE TITLE
Measurement metrics dependent on dataset sizes. 

### ISSUE THREAD

**Issue by ankushc48** [lifebit-ai/mlops#21]
> Without metrics scaling it is more easily to realise how the model is good.

@stefan-lifebit, Can you please explain in more details what does it mean? It does not make any sense to me. @ceru23, could you also please contribute, if you understand it as you approved it. Metrics scaling (as it may be called) is likely to help in the comparison when datasets sizes are updated. 
Dividing with constant number does not change anything in terms what is it telling you about the model at least for the metrics used in these scripts. Please, correct me wherever I am wrong. 

> Since the goal is to use the same dataset across all experiments that are compared with each other scaling does not have to be present.

As discussed multiple times, this dataset should be updated, considering the new samples which might be there due to [model shift and decay](https://towardsdatascience.com/concept-drift-and-model-decay-in-machine-learning-a98a809ea8d4). New changes we are making, for example: this [ bug fix](https://github.com/lifebit-ai/nlp/pull/139) we need to update eval set to add some examples which are covered by this bug fix might also be a good idea. So that we at least know what useful is coming out of this terms of accuracy. 

Let me know, if I am missing something.

_Originally posted by @ankushc48 in https://github.com/lifebit-ai/mlops/issues/20#issuecomment-1252059517_

**Comment by stefan-lifebit** [comment]
Yes I agree with you but the goal of this MLOps experiment tracking is to test always on the same dataset and whenever you change it the new calculations should be applied in order to compare different versions.

**Comment by ankushc48** [comment]
And what is the your rationale behind always keeping the same dataset? 

The aim of ML metrics is always to have a statistical measure which should not matter on which evaluation dataset size (or update it when we choose to) until it's a good representation of real world data. If we keep it the always the "same", we assume that we have a perfect dataset for testing, which is (and will never be) true. 

Also, it does not make any sense to have metrics dependent on dataset size. That's why [standard metrics](https://scikit-learn.org/stable/modules/classes.html#module-sklearn.metrics) never have something which is addition of all true positives, false positives etc. for the dataset. 

You can have precision, recall here, instead of total TP, FP etc. to keep it closer to how the standard metrics are calculated.

**Comment by stefan-lifebit** [comment]
I am not saying that we will always have the same dataset, but the moment we change it we have to run the calculations all over again

**Comment by ankushc48** [comment]
Okay, never mind scaling, I also realised it was a weird idea. 

But comparing total TP (and other two things) in two datasets also does not make any sense. So I would strongly recommend to change it to precision and recall.

**Comment by ceru23** [comment]
yes you are right ankush, I overlooked the true positive, and I agree it is better to move it to precision and recall, in addition to the f1-score

**Comment by stefan-lifebit** [comment]
I will set up a call with you tomorrow to explain everything
